tests/tests_default.yml

```diff
@@ -2,6 +2,54 @@
 ---
 - name: Ensure that the role runs with default parameters
   hosts: all
-  gather_facts: false  # test that role works in this case
-  roles:
-    - linux-system-roles.template
+  gather_facts: true  # test that role works in this case
+  become: yes
+  become_user: root
+
+  tasks:
+    - name: Test default settings
+      block:
+        - name: Run postgresql role
+          include_role:
+            name: linux-system-roles.postgresql
+          vars:
+            postgresql_version: "13"
+
+        - meta: flush_handlers
+
+        - name: test - postgresql-server running
+          command: systemctl is-active postgresql
+          changed_when: false
+
+        - name: test - postgresql-server is enabled
+          command: systemctl is-enabled postgresql
+          changed_when: false
+
+        - name: test - database is accesible for super user usign Unix socket
+          become: yes
+          become_user: postgres
+          shell: echo '\q' | psql
+          async: 3             # in case of password promt we need to fail
+          changed_when: false
+
+        - name: check - server tunning is used - shared buffers
+          become: yes
+          become_user: postgres
+          shell: echo "SHOW shared_buffers;" | psql
+          register: result
+          changed_when: false
+
+        - name: test - server tunning is used - shared buffers
+          assert:
+            that: "{{ (ansible_memory_mb.real.total/4)|int|abs  }}|string  in result.stdout"
+
+        - name: check - server tunning is used - effective cache size
+          become: yes
+          become_user: postgres
```

Comment:
you can group tasks that do `become` to a single block like this:
```
        - name: test- become postgresql and test
          become: yes
          become_user: postgres
          block:
            - name: test - db is accessible for super user using Unix socket
              shell: echo '\q' | psql
              async: 3             # in case of password promt we need to fail
              changed_when: false
```